src/OrchardCore.Modules/OrchardCore.ContentFields/Extensions/ContentPartFieldDefinitionExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
```

Comment:
Just some thought here. Could we do the same but without using Newtonsoft.Json and replace it with System.Text.Json?

Reply:
The `ContentItem.Content` points to `OrchardCore.ContentManagement.ContentElement.Data` which is a Newtonsoft `JObject`. This class doesn't do any serializing, just access this existing property so trying to use  System.Text.Json here would require conversion. Better table it until `ContentElement` itself is ported to STJ, I think.

Reply:
Ok, forget it for now then.

Reply:
Will Linq be concern here for allocation and performance?

Reply:
We pass on the enumerables without using an intermediary list like before, so that should improve allocation. I'm not sure about the performance, but we don't do anything complex so it should be fine. Also these providers are only called when you update a content item's field; that's relatively rare so you shouldn't be concerned.